### PR TITLE
Use correct filemode when creating local directories

### PR DESCRIPTION
**Pull request by agclever**
`os.ModeDir` is actually a mask for the directory bit of the file mode. The previous code created directories with permissions `0000` which meant that nobody had read-write permissions to the directory. Instead we set the filemode to `0700` which means `drwx------`.

Tested:
ran locally by modifying vendored code.

### CODE REVIEW

File: pathio.go [hunk 1]
```diff
@@ -122,7 +122,7 @@ func writeToS3(s3Conn s3Connection, input io.ReadSeeker, disableEncryption bool)
 
 // writeToLocalFile writes the given file locally
 func writeToLocalFile(path string, input io.ReadSeeker) error {
-	if err := os.MkdirAll(filepath.Dir(path), os.ModeDir); err != nil {
+	if err := os.MkdirAll(filepath.Dir(path), 0700); err != nil {
```
Comment (kvigen) [inline pathio.go]: is there no constant for this?

**Comment by eighnjel** [comment]
Interesting, curious to know how to catch that in the last pr. but sounds like it's working now, so lgtm

File: pathio.go (hunk 1 above)
Comment (agclever) [inline pathio.go]: I couldn't find one. Even the golang source files use the same approach. https://golang.org/src/io/ioutil/tempfile.go?s=2143:2200#L84

**Comment by agclever** [comment]
@eighnjel I don't see a way other than actually writing to a local file to create a directory. We could have had a simple test for write under local case which would have caught this in case the path had a missing directory.

**Pull request merged**